expression/builtin_string_vec.go

```diff
@@ -135,3 +135,33 @@ func (b *builtinStringIsNullSig) vecEvalInt(input *chunk.Chunk, result *chunk.Co
 func (b *builtinStringIsNullSig) vectorized() bool {
 	return true
 }
+
+// vecEvalString evals a REVERSE(str).
+// See https://dev.mysql.com/doc/refman/8.0/en/string-functions.html#function_reverse
+func (b *builtinReverseSig) vecEvalString(input *chunk.Chunk, result *chunk.Column) error {
+	n := input.NumRows()
+	buf, err := b.bufAllocator.get(types.ETString, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf)
+	if err := b.args[0].VecEvalString(b.ctx, input, buf); err != nil {
```

Comment:
We can directly use the `result` variable to store the result of `args[0]`. Thus we can improve the performance by saving a memory copy.

Reply:
I have modified, PTAL @zz-jason